server/schedule/placement/rule_manager.go

```diff
@@ -41,13 +41,15 @@ type RuleManager struct {
 	ruleList    ruleList
 
 	// used for rule validation
-	keyType string
+	keyType   string
+	getStores func() []*core.StoreInfo
 }
 
 // NewRuleManager creates a RuleManager instance.
-func NewRuleManager(store *core.Storage) *RuleManager {
+func NewRuleManager(store *core.Storage, getStores func() []*core.StoreInfo) *RuleManager {
 	return &RuleManager{
 		store:      store,
```

Comment:
Maybe we need to rename this to `storage` to avoid ambiguity?

Reply:
Done.